cmd/private_network_create.go

```diff
@@ -48,37 +50,85 @@ func (c *privateNetworkCreateCmd) cmdPreRun(cmd *cobra.Command, args []string) e
 }
 
 func (c *privateNetworkCreateCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, c.Zone))
+	ctx := gContext
+	client, err := switchClientZoneV3(ctx, globalstate.EgoscaleV3Client, c.Zone)
+	if err != nil {
+		return err
+	}
 
-	privateNetwork := &egoscale.PrivateNetwork{
-		Description: utils.NonEmptyStringPtr(c.Description),
-		EndIP: func() (v *net.IP) {
+	req := v3.CreatePrivateNetworkRequest{
+		Description: func() string {
+			if c.Description != "" {
+				return *utils.NonEmptyStringPtr(c.Description)
+			}
+			return ""
+		}(),
+		EndIP: func() net.IP {
 			if c.EndIP != "" {
-				ip := net.ParseIP(c.EndIP)
-				v = &ip
+				return net.ParseIP(c.EndIP)
 			}
-			return
+			return nil
 		}(),
```

Comment:
Here you can also return directly `net.ParseIP(c.EndIP)` parse IP will return nil on invalid ;)